kernel/src/hil/crc.rs

```diff
@@ -0,0 +1,35 @@
+// Generic interface for CRC computation
+
+use returncode::ReturnCode;
+
+#[derive(Copy, Clone)]
+pub enum Polynomial {
+	CCIT8023,   // Polynomial 0x04C11DB7
+	CASTAGNOLI, // Polynomial 0x1EDC6F41
+	CCIT16,     // Polynomial 0x1021
+}
+
+pub fn poly_from_int(i: usize) -> Option<Polynomial> {
+    match i {
+        0 => Some(Polynomial::CCIT8023),
```

Comment:
These is kind of a choice of the system call driver, right? I think this function should be there, then, instead.